### ISSUE TITLE
Cannot disable 'cyclic-import' warning

### ISSUE THREAD

**Issue by rpullm168** [pylint-dev/pylint#9490]
### Bug description

```python
Neither:
`# pylint: disable=cyclic-import`
nor:
`# pylint: disable=R0401`

squelches the "Cyclic import" warning.  Tried at all scopes, all the way up to global module scope, to no avail: warning is still emitted.

(The apparent cyclic import is a false positive that does not actually occur in runtime, which is not the bug I'm reporting.)

The warning can be successfully disabled by adding a `disable=cyclic-import` clause in the `.pylintrc` file, but that suppresses this check for all modules in the package directory, not desirable.
```

### Command used

```shell
pylint setup.py
```

### Pylint output

```python
************* Module setup
setup.py:1:0: R0401: Cyclic import (ea_vault.vault_api -> ea_vault.vault_cache) (cyclic-import)
```

### Expected behavior

(no warning)

### Pylint version

```shell
pylint 3.1.0
astroid 3.1.0
Python 3.12.2 (c3.12.2:6abddd9f6a, Feb 6, 2024, 17:02:06) [Clang 13.0.0 (clang-1200.0.29.30)]
```

### OS / Environment

macOS Darwin Kernel Version 21.3.0

**Comment by Pierre-Sassoulas** [comment]
Related to #6983 and #850, we still need to permit to break cyclic-import with a local disable.
